packages/odf/components/topology/sidebar/TopologySideBarContent.tsx

```diff
@@ -38,103 +36,153 @@ type TopologySideBarContentProps = {
-          resource.metadata.name
-        );
-      let alertsFilter = defaultAlertsFilter;
-      let [detailsTab, resourcesTab, observeTab] = Array(3)
-        .fill(null)
-        .map(() => <DataUnavailableError />);
-      switch (resource.kind) {
-        case NodeModel.kind:
-          alertsFilter = (alert: Alert) =>
-            alert?.labels?.node === resource.metadata.name ||
-            defaultAlertsFilter(alert);
-          detailsTab = <NodeDetails node={resource as NodeKind} />;
-          resourcesTab = <NodeResources node={resource as NodeKind} />;
-          observeTab = <NodeObserve node={resource as NodeKind} />;
-          break;
-        case DeploymentModel.kind:
-          detailsTab = <DeploymentDetails resource={resource as any} />;
-          resourcesTab = <DeploymentResources deployment={resource as any} />;
-          observeTab = <DeploymentObserve deployment={resource as any} />;
-          break;
-        case StorageClusterModel.kind:
-          detailsTab = (
-            <StorageClusterDetails
-              storageCluster={resource as StorageClusterKind}
-            />
-          );
-          resourcesTab = <StorageClusterResources />;
-          observeTab = <StorageClusterObserve />;
-          break;
-      }
-      if (!alertsFilter) {
-        alertsFilter = defaultAlertsFilter;
-      }
-      const alertsTab = <AlertsDetails alertsFilter={alertsFilter} />;
-      return [alertsTab, detailsTab, resourcesTab, observeTab];
-    }, [resource]);
-
-  const tabs: TabPage[] = React.useMemo(
-    () =>
-      !inFlight
-        ? [
-            {
-              title: t('Alerts'),
-              href: '#sidebar-alerts',
-              component: () => alertsTab,
-            },
-            {
-              title: t('Details'),
-              href: '#sidebar-details',
-              component: () => detailsTab,
-            },
-            {
-              title: t('Resources'),
-              href: '#sidebar-resources',
-              component: () => resourcesTab,
-            },
-            {
-              title: t('Observe'),
-              href: '#sidebar-observe',
-              component: () => observeTab,
-            },
-          ]
-        : [],
-    [t, inFlight, alertsTab, detailsTab, resourcesTab, observeTab]
+
+  const onSelect = React.useCallback(
+    (event, tabIndex) => {
+      setActiveTab(tabIndex);
+    },
+    [setActiveTab]
   );
 
   const title = <DetailsPageTitle resource={resource} resourceModel={model} />;
 
-  return (
+  return !inFlight ? (
     <>
       <PageHeading
         title={title}
         resource={resource}
         getResourceStatus={getResourceStatus}
         className="odf-topology-sidebar-heading"
       />
-      {tabs.length > 0 ? (
-        <Tabs id="odf-topology-sidebar-tab-container" tabs={tabs} />
-      ) : (
-        <LoadingBox />
-      )}
+      <Tabs activeKey={activeTab} onSelect={onSelect} unmountOnExit>
```

Comment:
@bipuladh Tested locally: we get the same tabs Look'N'Feel (for consistency) by adding `odf-tabs` class
`import '@odf/shared/utils/tabs.scss';`

```suggestion
      <Tabs className="odf-tabs" activeKey={activeTab} onSelect={onSelect} unmountOnExit>
```